### PR TITLE
[JAMES-3226] Automatically publish staging doc

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
```
org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
WorkflowScript: 163: unexpected token: } @ line 163, column 1.
   }
   ^

1 error

	at org.codehaus.groovy.control.ErrorCollector.failIfErrors(ErrorCollector.java:310)
	at org.codehaus.groovy.control.ErrorCollector.addFatalError(ErrorCollector.java:150)
	at org.codehaus.groovy.control.ErrorCollector.addError(ErrorCollector.java:120)
	at org.codehaus.groovy.control.ErrorCollector.addError(ErrorCollector.java:132)
	at org.codehaus.groovy.control.SourceUnit.addError(SourceUnit.java:350)
	at org.codehaus.groovy.antlr.AntlrParserPlugin.transformCSTIntoAST(AntlrParserPlugin.java:144)
	at org.codehaus.groovy.antlr.AntlrParserPlugin.parseCST(AntlrParserPlugin.java:110)
	at org.codehaus.groovy.control.SourceUnit.parse(SourceUnit.java:234)
	at org.codehaus.groovy.control.CompilationUnit$1.call(CompilationUnit.java:168)
	at org.codehaus.groovy.control.CompilationUnit.applyToSourceUnits(CompilationUnit.java:943)
	at org.codehaus.groovy.control.CompilationUnit.doPhaseOperation(CompilationUnit.java:605)
	at org.codehaus.groovy.control.CompilationUnit.processPhaseOperations(CompilationUnit.java:581)
	at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:558)
	at groovy.lang.GroovyClassLoader.doParseClass(GroovyClassLoader.java:298)
	at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:268)
	at groovy.lang.GroovyShell.parseClass(GroovyShell.java:688)
	at groovy.lang.GroovyShell.parse(GroovyShell.java:700)
	at org.jenkinsci.plugins.workflow.cps.CpsGroovyShell.doParse(CpsGroovyShell.java:142)
	at org.jenkinsci.plugins.workflow.cps.CpsGroovyShell.reparse(CpsGroovyShell.java:127)
	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.parseScript(CpsFlowExecution.java:571)
	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.start(CpsFlowExecution.java:523)
	at org.jenkinsci.plugins.workflow.job.WorkflowRun.run(WorkflowRun.java:337)
	at hudson.model.ResourceController.execute(ResourceController.java:97)
	at hudson.model.Executor.run(Executor.java:429)
Finished: FAILURE
```

**Comment by chibenwa** [comment]
```
org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
WorkflowScript: 122: Undefined section "unstable" @ line 122, column 5.
       unstable {
       ^

WorkflowScript: 127: Undefined section "success" @ line 127, column 5.
       success {
       ^

WorkflowScript: 159: Undefined section "always" @ line 159, column 5.
       always {
       ^
```

I think `unstable` sction makes no sense here.

**Comment by jeantil** [comment]
so no it doesn't make much sense but it seems like it's not the issue here success and always should be defined but they don't seem to work either

**Comment by chibenwa** [comment]
Yes I do not understand either.

**Comment by jeantil** [comment]
it a `{}` thing again you need to revert a26e00b and to remove the } after the failure block (around line 117)

File: Jenkinsfile [hunk 1]
```diff
@@ -88,45 +92,71 @@ pipeline {
...
+                    )
+                }
             }
-        }
 
-        // If this build didn't fail, but there were failing tests, send an email to the list.
-        unstable {
-            echo "Unstable "
         }
+    }
```
Comment (jeantil) [inline Jenkinsfile]: that's the one to remove after restoring the one at the bottom ! here it incorrectly closes the post scope which should englobe all the status filters (failure, unstable, always, success)

**Comment by jeantil** [comment]
I tried linting with https://docs.cloudbees.com/docs/admin-resources/latest/pipelines/pipeline-dev-tools#linter but for now I couldn't get it to work :(

**Comment by jeantil** [comment]
I pushed a fixed version on my branch (with separated commits for each step) , I was able to lint it 
```
export JENKINS_URL=https://builds.apache.org/ 
export  JENKINS_CRUMB=$(http $JENKINS_URL/crumbIssuer/api/xml\?xpath=concat\(//crumbRequestField,\":\",//crumb\))   curl -X POST -H $JENKINS_CRUMB -F "jenkinsfile=<Jenkinsfile" $JENKINS_URL/pipeline-model-converter/validate
# returns Jenkinsfile successfully validated.
```
it lives at https://github.com/jeantil/james-site/tree/automatically-publish-staging-documenttion-on-james-project-build

**Comment by jeantil** [comment]
\o/ :rocket:

**Comment by jeantil** [comment]
Quick update that this configuration does work : https://builds.apache.org/job/james/job/ApacheJames-Website/job/staging/13/ was triggered by upstream https://builds.apache.org/job/james/job/ApacheJames/job/master/57/

we have auto deploy of documentation work :tada:

**Pull request merged**